src/script/interpreter.h

```diff
@@ -161,6 +161,18 @@ class BaseSignatureChecker
     virtual ~BaseSignatureChecker() {}
 };
 
+/** A low level signature checker.  Note that the caller must verify that the hash passed in relates to a known message (unlike for the other signature checkers). */
+class SimpleSignatureChecker : public BaseSignatureChecker
+{
+private:
+    uint256 hash;
+
+public:
+    const uint256& GetHash() const { return hash; }
+    explicit SimpleSignatureChecker(const uint256& hash_in) : hash(hash_in) {}
```

Comment:
nit: Move constructor one line up, immediately after `public` (like for the other class `SimpleSignatureCreator` and for most classes in general I'd say)

Reply:
I think it's fine as is, and there are plenty of examples of the constructor not being right below `public:`.